airflow/providers/databricks/sensors/databricks.py

```diff
@@ -0,0 +1,231 @@
+        client_parameters: dict[str, Any] | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.databricks_conn_id = databricks_conn_id
+        self._http_path = http_path
+        self._sql_endpoint_name = sql_endpoint_name
+        self.session_config = session_configuration
+        self.http_headers = http_headers
+        self.catalog = catalog
+        self.schema = "default" if not schema else schema
+        self.table_name = table_name
+        self.partition_name = partition_name
+        self.db_sensor_type = db_sensor_type
+        self.timestamp = timestamp
+        self.caller = caller
+        self.client_parameters = client_parameters or {}
+        self.hook_params = kwargs.pop("hook_params", {})
+        self.handler = handler
+
+    def _get_hook(self) -> DatabricksSqlHook:
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            self._http_path,
+            self._sql_endpoint_name,
+            self.session_config,
+            self.http_headers,
+            self.catalog,
+            self.schema,
+            self.caller,
+            **self.client_parameters,
+            **self.hook_params,
+        )
+
+    def _generic_sql_sensor(self, sql):
+        """Runs SQL commands in Databricks SQL Warehouse.
+
+        Args:
+            sql (_type_): SQL query
+
+        Returns:
+            Any: SQL query results, mixed type.
+        """
+        hook = self._get_hook()
+        sql_result = hook.run(
+            sql,
+            handler=self.handler if self.do_xcom_push else None,
+        )
+        return sql_result
+
+    @staticmethod
+    def get_previous_version(context: Context, lookup_key):
+        """Gets previous version of the table stored in Airflow metadata.
+
+        Args:
+            context (Context): Airflow context
+            lookup_key (_type_): Unique lookup key used to store values related to a specific table.
+
+        Returns:
+            int: Version number
+        """
+        return context["ti"].xcom_pull(key=lookup_key, include_prior_dates=True)
+
+    @staticmethod
+    def set_version(context: Context, lookup_key, version):
+        """Sets a specific version number for a Databricks table to the Airflow metadata using
+        an existing lookup key.
+
+        Args:
+            context (Context): Airflow context
+            lookup_key (_type_): Unique lookup key used to store values related to a specific table.
+            version (int): Version number
+        """
+        context["ti"].xcom_push(key=lookup_key, value=version)
+
+    def get_current_table_version(self, table_name, time_range):
+        change_sql = (
+            f"SELECT COUNT(version) as versions from "
+            f"(DESCRIBE HISTORY {table_name}) "
+            f"WHERE timestamp >= '{time_range}'"
+        )
+        result = self._generic_sql_sensor(change_sql)[0][0]
+        return result
```

Comment:
It's better to move this into a separate sensor class

Reply:
Moved it to separate classes, creating 3 - one for each sensor.